In [1]:
#import module yang dibutuhkan
import numpy as np
import keras
import pandas as pd
import tensorflow as tf
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report

In [2]:
!pip install -q -U keras-tuner

     |████████████████████████████████| 133 kB 8.5 MB/s 


In [3]:
import keras_tuner as kt

In [4]:
#upload file
from google.colab import files
upload = files.upload()

Saving Jamur.csv to Jamur.csv


#Pengecekan Data

In [5]:
#melihat data
df = pd.read_csv('Jamur.csv')
df.head()

class cap-shape cap-surface cap-color bruises odor gill-attachment  \
0     p         x           s         n       t    p               f   
1     e         x           s         y       t    a               f   
2     e         b           s         w       t    l               f   
3     p         x           y         w       t    p               f   
4     e         x           s         g       f    n               f   

  gill-spacing gill-size gill-color  ... stalk-surface-below-ring  \
0            c         n          k  ...                        s   
1            c         b          k  ...                        s   
2            c         b          n  ...                        s   
3            c         n          n  ...                        s   
4            w         b          k  ...                        s   

  stalk-color-above-ring stalk-color-below-ring veil-type veil-color  \
0                      w                      w         p          w   
1                      w                      w         p          w   
2                      w                      w         p          w   
3                      w                      w         p          w   
4                      w                      w         p          w   

  ring-number ring-type spore-print-color population habitat  
0           o         p                 k          s       u  
1           o         p                 n          n       g  
2           o         p                 n          n       m  
3           o         p                 k          s       u  
4           o         e                 n          a       g  

[5 rows x 23 columns]

In [6]:
#melihat kolom
df.columns

Index(['class', 'cap-shape', 'cap-surface', 'cap-color', 'bruises', 'odor',
       'gill-attachment', 'gill-spacing', 'gill-size', 'gill-color',
       'stalk-shape', 'stalk-root', 'stalk-surface-above-ring',
       'stalk-surface-below-ring', 'stalk-color-above-ring',
       'stalk-color-below-ring', 'veil-type', 'veil-color', 'ring-number',
       'ring-type', 'spore-print-color', 'population', 'habitat'],
      dtype='object')

In [7]:
#melihat nilai apa saja yang ada pada setiap kolom
df["class"].unique()

array(['p', 'e'], dtype=object)

In [8]:
df["cap-shape"].unique()

array(['x', 'b', 's', 'f', 'k', 'c'], dtype=object)

In [9]:
df["cap-surface"].unique()

array(['s', 'y', 'f', 'g'], dtype=object)

In [10]:
df["cap-color"].unique()

array(['n', 'y', 'w', 'g', 'e', 'p', 'b', 'u', 'c', 'r'], dtype=object)

In [11]:
df["bruises"].unique()

array(['t', 'f'], dtype=object)

In [12]:
df["odor"].unique()

array(['p', 'a', 'l', 'n', 'f', 'c', 'y', 's', 'm'], dtype=object)

In [13]:
df["gill-attachment"].unique()

array(['f', 'a'], dtype=object)

In [14]:
df["gill-spacing"].unique()

array(['c', 'w'], dtype=object)

In [15]:
df["gill-size"].unique()

array(['n', 'b'], dtype=object)

In [16]:
df["gill-color"].unique()

array(['k', 'n', 'g', 'p', 'w', 'h', 'u', 'e', 'b', 'r', 'y', 'o'],
      dtype=object)

In [17]:
df["stalk-shape"].unique()

array(['e', 't'], dtype=object)

In [18]:
df["stalk-root"].unique() #ada missing value disini

array(['e', 'c', 'b', 'r', '?'], dtype=object)

In [19]:
df["stalk-surface-above-ring"].unique()

array(['s', 'f', 'k', 'y'], dtype=object)

In [20]:
df["stalk-surface-below-ring"].unique()

array(['s', 'f', 'y', 'k'], dtype=object)

In [21]:
df["veil-type"].unique() #trivial

array(['p'], dtype=object)

In [22]:
df["veil-color"].unique()

array(['w', 'n', 'o', 'y'], dtype=object)

In [23]:
df["ring-number"].unique()

array(['o', 't', 'n'], dtype=object)

In [24]:
df["ring-type"].unique()

array(['p', 'e', 'l', 'f', 'n'], dtype=object)

In [25]:
df["spore-print-color"].unique()

array(['k', 'n', 'u', 'h', 'w', 'r', 'o', 'y', 'b'], dtype=object)

In [26]:
df["population"].unique()

array(['s', 'n', 'a', 'v', 'y', 'c'], dtype=object)

In [27]:
df["habitat"].unique()

array(['u', 'g', 'm', 'd', 'p', 'w', 'l'], dtype=object)

In [28]:
#cek missing value
df.isnull().sum()

class                       0
cap-shape                   0
cap-surface                 0
cap-color                   0
bruises                     0
odor                        0
gill-attachment             0
gill-spacing                0
gill-size                   0
gill-color                  0
stalk-shape                 0
stalk-root                  0
stalk-surface-above-ring    0
stalk-surface-below-ring    0
stalk-color-above-ring      0
stalk-color-below-ring      0
veil-type                   0
veil-color                  0
ring-number                 0
ring-type                   0
spore-print-color           0
population                  0
habitat                     0
dtype: int64

In [29]:
#melihat gambaran data
df.describe()

class cap-shape cap-surface cap-color bruises  odor gill-attachment  \
count   8124      8124        8124      8124    8124  8124            8124   
unique     2         6           4        10       2     9               2   
top        e         x           y         n       f     n               f   
freq    4208      3656        3244      2284    4748  3528            7914   

       gill-spacing gill-size gill-color  ... stalk-surface-below-ring  \
count          8124      8124       8124  ...                     8124   
unique            2         2         12  ...                        4   
top               c         b          b  ...                        s   
freq           6812      5612       1728  ...                     4936   

       stalk-color-above-ring stalk-color-below-ring veil-type veil-color  \
count                    8124                   8124      8124       8124   
unique                      9                      9         1          4   
top                         w                      w         p          w   
freq                     4464                   4384      8124       7924   

       ring-number ring-type spore-print-color population habitat  
count         8124      8124              8124       8124    8124  
unique           3         5                 9          6       7  
top              o         p                 w          v       d  
freq          7488      3968              2388       4040    3148  

[4 rows x 23 columns]

#Data Cleaning

In [30]:
#hapus veil-type pada data, karena trivial
df = df[['class', 'cap-shape', 'cap-surface', 'cap-color', 'bruises', 'odor','gill-attachment', 'gill-spacing', 'gill-size', 'gill-color', 'stalk-shape', 'stalk-root', 'stalk-surface-above-ring','stalk-surface-below-ring', 'stalk-color-above-ring','stalk-color-below-ring', 'veil-color', 'ring-number','ring-type', 'spore-print-color', 'population', 'habitat']]
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8124 entries, 0 to 8123
Data columns (total 22 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   class                     8124 non-null   object
 1   cap-shape                 8124 non-null   object
 2   cap-surface               8124 non-null   object
 3   cap-color                 8124 non-null   object
 4   bruises                   8124 non-null   object
 5   odor                      8124 non-null   object
 6   gill-attachment           8124 non-null   object
 7   gill-spacing              8124 non-null   object
 8   gill-size                 8124 non-null   object
 9   gill-color                8124 non-null   object
 10  stalk-shape               8124 non-null   object
 11  stalk-root                8124 non-null   object
 12  stalk-surface-above-ring  8124 non-null   object
 13  stalk-surface-below-ring  8124 non-null   object
 14  stalk-color-above-ring  

In [31]:
#hapus semua baris yang stalk-root = ? (missing value)
df = df[df["stalk-root"]!= '?']
df.head()

class cap-shape cap-surface cap-color bruises odor gill-attachment  \
0     p         x           s         n       t    p               f   
1     e         x           s         y       t    a               f   
2     e         b           s         w       t    l               f   
3     p         x           y         w       t    p               f   
4     e         x           s         g       f    n               f   

  gill-spacing gill-size gill-color  ... stalk-surface-above-ring  \
0            c         n          k  ...                        s   
1            c         b          k  ...                        s   
2            c         b          n  ...                        s   
3            c         n          n  ...                        s   
4            w         b          k  ...                        s   

  stalk-surface-below-ring stalk-color-above-ring stalk-color-below-ring  \
0                        s                      w                      w   
1                        s                      w                      w   
2                        s                      w                      w   
3                        s                      w                      w   
4                        s                      w                      w   

  veil-color ring-number ring-type spore-print-color population habitat  
0          w           o         p                 k          s       u  
1          w           o         p                 n          n       g  
2          w           o         p                 n          n       m  
3          w           o         p                 k          s       u  
4          w           o         e                 n          a       g  

[5 rows x 22 columns]

In [32]:
#cek jumlah data
df.shape

(5644, 22)

In [33]:
df.columns

Index(['class', 'cap-shape', 'cap-surface', 'cap-color', 'bruises', 'odor',
       'gill-attachment', 'gill-spacing', 'gill-size', 'gill-color',
       'stalk-shape', 'stalk-root', 'stalk-surface-above-ring',
       'stalk-surface-below-ring', 'stalk-color-above-ring',
       'stalk-color-below-ring', 'veil-color', 'ring-number', 'ring-type',
       'spore-print-color', 'population', 'habitat'],
      dtype='object')

#Preprocessing Data

In [34]:
#mengubah data objek menjadi numerik

from sklearn.preprocessing import LabelEncoder

labelencoder = LabelEncoder()
df['class'] = labelencoder.fit_transform(df['class'])
df['cap-shape'] = labelencoder.fit_transform(df['cap-shape'])
df['cap-surface'] = labelencoder.fit_transform(df['cap-surface'])
df['cap-color'] = labelencoder.fit_transform(df['cap-color'])
df['bruises'] = labelencoder.fit_transform(df['bruises'])
df['odor'] = labelencoder.fit_transform(df['odor'])
df['gill-attachment'] = labelencoder.fit_transform(df['gill-attachment'])
df['gill-spacing'] = labelencoder.fit_transform(df['gill-spacing'])
df['gill-size'] = labelencoder.fit_transform(df['gill-size'])
df['gill-color'] = labelencoder.fit_transform(df['gill-color'])
df['stalk-shape'] = labelencoder.fit_transform(df['stalk-shape'])
df['stalk-root'] = labelencoder.fit_transform(df['stalk-root'])
df['stalk-surface-above-ring'] = labelencoder.fit_transform(df['stalk-surface-above-ring'])
df['stalk-surface-below-ring'] = labelencoder.fit_transform(df['stalk-surface-below-ring'])
df['stalk-color-above-ring'] = labelencoder.fit_transform(df['cap-surface'])
df['stalk-color-below-ring'] = labelencoder.fit_transform(df['cap-surface'])
df['veil-color'] = labelencoder.fit_transform(df['veil-color'])
df['ring-number'] = labelencoder.fit_transform(df['ring-number'])
df['ring-type'] = labelencoder.fit_transform(df['ring-type'])
df['spore-print-color'] = labelencoder.fit_transform(df['spore-print-color'])
df['population'] = labelencoder.fit_transform(df['population'])
df['habitat'] = labelencoder.fit_transform(df['habitat'])

df.head()

class  cap-shape  cap-surface  cap-color  bruises  odor  gill-attachment  \
0      1          5            2          4        1     6                1   
1      0          5            2          7        1     0                1   
2      0          0            2          6        1     3                1   
3      1          5            3          6        1     6                1   
4      0          5            2          3        0     5                1   

   gill-spacing  gill-size  gill-color  ...  stalk-surface-above-ring  \
0             0          1           2  ...                         2   
1             0          0           2  ...                         2   
2             0          0           3  ...                         2   
3             0          1           3  ...                         2   
4             1          0           2  ...                         2   

   stalk-surface-below-ring  stalk-color-above-ring  stalk-color-below-ring  \
0                         2                       2                       2   
1                         2                       2                       2   
2                         2                       2                       2   
3                         2                       3                       3   
4                         2                       2                       2   

   veil-color  ring-number  ring-type  spore-print-color  population  habitat  
0           0            1          3                  1           3        5  
1           0            1          3                  2           2        1  
2           0            1          3                  2           2        3  
3           0            1          3                  1           3        5  
4           0            1          0                  2           0        1  

[5 rows x 22 columns]

In [35]:
#cek korelasi awal
print(df.corr())

                             class  cap-shape  cap-surface  cap-color  \
class                     1.000000   0.053155     0.046859   0.150741   
cap-shape                 0.053155   1.000000    -0.068688  -0.056421   
cap-surface               0.046859  -0.068688     1.000000   0.093355   
cap-color                 0.150741  -0.056421     0.093355   1.000000   
bruises                  -0.435562  -0.097782     0.228590  -0.165794   
odor                     -0.455566   0.000011    -0.108697  -0.400652   
gill-attachment          -0.071945   0.002963    -0.058104   0.059693   
gill-spacing             -0.264160   0.051432    -0.204448   0.062189   
gill-size                 0.215289   0.103443    -0.042356   0.102676   
gill-color               -0.318339  -0.006431     0.073668  -0.242099   
stalk-shape              -0.592446   0.049043    -0.173554  -0.444682   
stalk-root               -0.291026  -0.027136     0.067789   0.222370   
stalk-surface-above-ring -0.375910  -0.033516     0

Ambil data dengan 5 korelasi terbesar. yaitu, habitat, gill-size, population, cap-color, dan cap-shape

#Prepocessing Data

In [36]:
df1 = df[['habitat', 'gill-size', 'population', 'cap-color', 'cap-shape', 'class']]
df1.head()

habitat  gill-size  population  cap-color  cap-shape  class
0        5          1           3          4          5      1
1        1          0           2          7          5      0
2        3          0           2          6          0      0
3        5          1           3          6          5      1
4        1          0           0          3          5      0

In [37]:
df1.describe()

habitat    gill-size   population    cap-color    cap-shape  \
count  5644.000000  5644.000000  5644.000000  5644.000000  5644.000000   
mean      1.236003     0.124734     3.712970     4.272856     3.420269   
std       1.597981     0.330447     1.328741     1.838018     1.659641   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     3.000000     3.000000     2.000000   
50%       1.000000     0.000000     4.000000     4.000000     5.000000   
75%       1.000000     0.000000     5.000000     6.000000     5.000000   
max       5.000000     1.000000     5.000000     7.000000     5.000000   

             class  
count  5644.000000  
mean      0.381999  
std       0.485919  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       1.000000  
max       1.000000

In [38]:
#bagi fitur yang dipilih dengan oleh nilai maksimumnya (proses skaling)
df1["habitat"], df1["population"], df1["cap-color"], df1["cap-shape"] = df1["habitat"]/5, df1["population"]/5, df1["cap-color"]/7, df1["cap-shape"]/5
df1.head()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


habitat  gill-size  population  cap-color  cap-shape  class
0      1.0          1         0.6   0.571429        1.0      1
1      0.2          0         0.4   1.000000        1.0      0
2      0.6          0         0.4   0.857143        0.0      0
3      1.0          1         0.6   0.857143        1.0      1
4      0.2          0         0.0   0.428571        1.0      0

In [39]:
df1.columns

Index(['habitat', 'gill-size', 'population', 'cap-color', 'cap-shape',
       'class'],
      dtype='object')

In [40]:
#definisikan fitur-fitur yang akan kita olah
X = df1[['habitat', 'gill-size', 'population', 'cap-color', 'cap-shape']].values
X[0:5]

array([[1.        , 1.        , 0.6       , 0.57142857, 1.        ],
       [0.2       , 0.        , 0.4       , 1.        , 1.        ],
       [0.6       , 0.        , 0.4       , 0.85714286, 0.        ],
       [1.        , 1.        , 0.6       , 0.85714286, 1.        ],
       [0.2       , 0.        , 0.        , 0.42857143, 1.        ]])

In [41]:
y = df1[['class']].values

y[100:150]

array([[0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [1],
       [0],
       [0],
       [0],
       [0],
       [0],
       [1],
       [0],
       [1],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [1],
       [0],
       [0],
       [1],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0],
       [0]])

#Train/Tes Split

In [42]:
#pemilihan/pemisahan data latih dan data tes
X_train_full, X_test, y_train_full, y_test = train_test_split(X, y, test_size = 0.2, random_state = 4)

In [43]:
#tampilkan ukuran data latih dan testnya
print("X_train_full: ", X_train_full.shape)
print("y_train_full: ", y_train_full.shape)
print("X_test: ", X_test.shape)
print("y_test: ", y_test.shape)

X_train_full:  (4515, 5)
y_train_full:  (4515, 1)
X_test:  (1129, 5)
y_test:  (1129, 1)


In [44]:
#definisikan validasi data dan data latih baru bagian data latih sebelumnya
X_valid, X_train = X_train_full[:1129], X_train_full[1129:]
y_valid, y_train = y_train_full[:1129], y_train_full[1129:]

In [45]:
#cek kelas apa saja nantinya
df1["class"].unique()

array([1, 0])

In [46]:
#definisikan kelas
class_jamur = [0,1]

#Modelling

In [47]:
#membangun model klasifikasi dengan hyperparameter
def build_model(hp):
  model = keras.models.Sequential()
  hp_layers = hp.Choice('layer', [2,3]) #banyak hidden layer 2 atau 3
  hp_neurons = hp.Int('neuron', min_value = 100, max_value=300, step = 100) #banyak neuron disetiap layer 100, 20

  for layer in range(hp_layers):
    model.add(keras.layers.Dense(hp_neurons, activation = 'relu')) #hidden layer
  model.add(keras.layers.Dense(10, activation = 'softmax')) #output layer 

  optimizer = tf.keras.optimizers.Adam()
  model.compile(loss = 'sparse_categorical_crossentropy', optimizer = optimizer, metrics = ['accuracy'])

  return model

In [48]:
tuner = kt.Hyperband(build_model, objective = 'val_accuracy', max_epochs = 10, directory = 'my_dir', project_name = 'my_tuner')

In [49]:
stop_early = tf.keras.callbacks.EarlyStopping(monitor = 'val_loss',patience = 5)

In [50]:
tuner.search(X_train, y_train, epochs = 50, validation_data = [X_valid, y_valid], callbacks = [stop_early])

Trial 6 Complete [00h 00m 01s]
val_accuracy: 0.8210806250572205

Best val_accuracy So Far: 0.8945969939231873
Total elapsed time: 00h 00m 11s
INFO:tensorflow:Oracle triggered exit


In [51]:
best_hps = tuner.get_best_hyperparameters()[0]
print("best number of layers: ", best_hps.get('layer'))
print("best number of neurons: ", best_hps.get('neuron'))

best number of layers:  3
best number of neurons:  300


In [52]:
model = tuner.hypermodel.build(best_hps)

In [53]:
model.fit(X_train, y_train, epochs = 150, validation_data = [X_valid, y_valid])

Epoch 1/150
106/106 [==============================] - 2s 10ms/step - loss: 0.6847 - accuracy: 0.7185 - val_loss: 0.4709 - val_accuracy: 0.8388
Epoch 2/150
106/106 [==============================] - 0s 4ms/step - loss: 0.4523 - accuracy: 0.8074 - val_loss: 0.3950 - val_accuracy: 0.8521
Epoch 3/150
106/106 [==============================] - 1s 5ms/step - loss: 0.3930 - accuracy: 0.8677 - val_loss: 0.3634 - val_accuracy: 0.9017
Epoch 4/150
106/106 [==============================] - 1s 5ms/step - loss: 0.3612 - accuracy: 0.8792 - val_loss: 0.3690 - val_accuracy: 0.8556
Epoch 5/150
106/106 [==============================] - 1s 5ms/step - loss: 0.3417 - accuracy: 0.8845 - val_loss: 0.2988 - val_accuracy: 0.9097
Epoch 6/150
106/106 [==============================] - 1s 5ms/step - loss: 0.2999 - accuracy: 0.9014 - val_loss: 0.2875 - val_accuracy: 0.9088
Epoch 7/150
106/106 [==============================] - 1s 5ms/step - loss: 0.3019 - accuracy: 0.9019 - val_loss: 0.2792 - val_accuracy: 0.909

#Evaluasi Model

In [54]:
#cek akurasi dan error
model.evaluate(X_test, y_test)

36/36 [==============================] - 0s 2ms/step - loss: 0.1318 - accuracy: 0.9486


[0.1318398267030716, 0.9486271142959595]

In [55]:
#hitung peluang apakah data masuk ke kelas 0 atau 1
y_proba = model.predict(X_test)
y_proba

array([[3.14098201e-04, 9.99685884e-01, 1.31929659e-14, ...,
        9.64886740e-15, 1.69975846e-14, 1.17498335e-14],
       [1.00000000e+00, 1.16958206e-23, 0.00000000e+00, ...,
        0.00000000e+00, 0.00000000e+00, 0.00000000e+00],
       [9.99885440e-01, 1.14535353e-04, 4.60869676e-10, ...,
        4.42031245e-10, 5.94535643e-10, 4.78113882e-10],
       ...,
       [9.99985933e-01, 1.40773291e-05, 3.26427100e-18, ...,
        1.06252293e-17, 6.56379273e-18, 7.58167114e-18],
       [9.99973059e-01, 2.69066950e-05, 2.24765950e-17, ...,
        6.79374885e-17, 4.42845633e-17, 5.01863448e-17],
       [7.92731047e-02, 9.20638859e-01, 1.12410116e-05, ...,
        1.13982296e-05, 1.12432617e-05, 1.08240947e-05]], dtype=float32)

In [56]:
#Hasil Prediksi
y_p = np.array([np.argmax(i) for i in y_proba]) #mendapatkan kelas
print(y_p)

[1 0 0 ... 0 0 1]


In [58]:
print(np.array(class_jamur)[y_p])

[1 0 0 ... 0 0 1]


In [59]:
#cek precision, recall, dan f-1 score
print(classification_report(y_test, y_p))

              precision    recall  f1-score   support

           0       0.94      0.98      0.96       716
           1       0.97      0.89      0.93       413

    accuracy                           0.95      1129
   macro avg       0.95      0.94      0.94      1129
weighted avg       0.95      0.95      0.95      1129



In [60]:
#fitur yang diduga kuat
print(df1.corr())

             habitat  gill-size  population  cap-color  cap-shape     class
habitat     1.000000   0.169763   -0.079260   0.196001  -0.057451  0.297412
gill-size   0.169763   1.000000   -0.008851   0.102676   0.103443  0.215289
population -0.079260  -0.008851    1.000000  -0.171096   0.048283  0.203882
cap-color   0.196001   0.102676   -0.171096   1.000000  -0.056421  0.150741
cap-shape  -0.057451   0.103443    0.048283  -0.056421   1.000000  0.053155
class       0.297412   0.215289    0.203882   0.150741   0.053155  1.000000


Algoritma:

#Pengecekan Data
1. Import module yang dibutuhkan
2. upload file dataset
3. Lakukan pengecekan data, apakah ada missing value atau tidak. Cek juga korelasi sebelum dilakukan pembersihan data
#Data Cleaning
4. Pembersihan data dari missing values dan data yang trivial
#Preprocessing Data
5. Mengubah data yang asalnya objek menjadi numerik
6. Cek korelasi setelah pembersihan data
7. Ambil 5 fitur dengan korelasi terbaik terhadap class. Artinya, pilih fitur yang lebih memiliki hubungan dengan atribut class
8. Bagi fitur yang terpilih oleh nilai maksimumnya. Tujuannya, agar datanya tidak memiliki rentang yang jauh sehingga dapat dilakukan proses klasifikasi dengan baik
9. Definisikan X sebagai kumpulan fitur yang terpilih dan y sebagai kelasnya
#Splitting Data
10. Definisikan pemilihan data training, test, dan validasi data
11. definsikan kelasnya, yaitu 0 dan 1.
#Modelling
12. Memodelkan klasifikasi dan menentukan hyperparameter terbaik supaya didapatkan model klasifikasi terbaik
#Evaluasi model
13. Mencari error, akurasi, f-1 score, precission, dan recall
#Analisis accuracy, precision, recall dan f1-score
didapatkan accuracy sebesar 0.95. Artinya, pemodelan klasifikasi ini sangat baik karena nilai akurasinya sudah sangat besar. precision, recall dan f1-score untuk kelas 0 berturut-turut sebesar 0.94, 0.98, dan 0.96 serta precision, recall dan f1-score untuk kelas 1 berturut-turut sebesar 0.97, 0.89, dan 0.93. Perhatikan bahwa dari nilai ketiganya, nilainya sudah sangat baik sekali. Sehingga, dapat kita simpulkan bahwa pemodelan kelas ini bekerja sangat baik.
#Fitur yang diduga kuat
Fitur yang diduga paling kuat adalah habitat dan gill-size karena korelasinya paling besar dari fitur yang lain. Artinya, habitat dan ukuran insang dari suatu jamur diduga kuat mempengaruhi apakah jamur tersebut termasuk pada kelas beracun atau tidak.